### PR TITLE
DEV-1742: Add lead error handling

### PR DESCRIPTION
**Pull request by ShiftyMcCool**
This adds better error handling for adding leads

ps-id: 7B5E4966-C42B-4E66-BEDA-A8B36A3E7407

### CODE REVIEW

File: src/app/leads/add-lead-page/add-lead-page.component.ts [hunk 1]
```diff
@@ -81,9 +81,9 @@ export class AddLeadPageComponent implements OnInit {
         catchError((errors) => {
           this.feedback.error({
             namespace: 'add-lead',
-            message: 'add-lead-error',
+            message: errors?.message ? errors.message : 'add-lead-error',
```
Comment (sierra-henderson) [inline src/app/leads/add-lead-page/add-lead-page.component.ts:84]: Is there any reason you are not using the submit method from the leads service here? Seems like unnecessary repitition of code

**Review by sierra-henderson — APPROVED** [review]
works, but I'd refactor the code to be more DRY

File: src/app/leads/add-lead-page/add-lead-page.component.ts (hunk 1 above)
Comment (ShiftyMcCool) [reply src/app/leads/add-lead-page/add-lead-page.component.ts:84]: @sierra-henderson you're right, it looks a bit repetitious but it's using a different GQL mutation.  I tried to dynamically switch between mutations but it didn't work out so well

Comment (sierra-henderson) [reply src/app/leads/add-lead-page/add-lead-page.component.ts:84]: This is g2g then!

**Pull request merged**